app/routes/erddap/buoy.js

```diff
@@ -80,26 +102,94 @@ router.get('/buoy', (req, res) => {
  *     responses:
  *       200:
  *         description: Success! New content is now available.
- * 
+ *
  */
 
-// Ex:  http://localhost:3004/erddap/buoy/coordinates?ids=bid2
+// Ex:  http://localhost:3004/erddap/buoy/coordinates?ids=bid2,bid3
 
-router.get('/buoy/coordinates', (req, res) => {
+router.get('/coordinates', (req, res) => {
   const ids = req.query.ids.split(',');
-  return Promise.all(getBuoysCoordinates({ids})).then(
-    (response) => {
-      const data = response.map((buoy) => {
-        return {
-          latitude: buoy.data.features[0].geometry.coordinates[1],
-          longitude: buoy.data.features[0].geometry.coordinates[0],
-          buoyId: buoy.data.features[0].properties.station_name
-        };
-      });
-      res.send(data);
-    }
-  );
-  
+  return Promise.all(getBuoysCoordinates({ ids })).then((response) => {
+    const data = response.map((buoy) => {
+      return {
+        latitude: buoy.data.features[0].geometry.coordinates[1],
+        longitude: buoy.data.features[0].geometry.coordinates[0],
+        buoyId: buoy.data.features[0].properties.station_name,
+        station_name: buoys.stationMap[buoy.data.features[0].properties.station_name]
+      };
+    });
+    res.send(data);
+  });
+});
+
+/**
+ * @swagger
+ * /erddap/buoy/summary:
+ *   get:
+ *     description: Get Buoy Coordinates from ERDDAP
+ *     parameters:
+ *      - name: end
+ *        in: query
+ *        description: End Date
+ *        required: true
+ *        type: string
+ *     responses:
+ *       200:
+ *         description: Success! New content is now available.
+ *
+ */
+
+// Ex:  http://localhost:3004/erddap/buoy/summary?end=2010-07-05T12:00:00Z
+
+router.get('/summary', cacheMiddleware, (req, res) => { 
+  const end = req.query.end;
+  const ids = buoys.ids;
+  const variable = buoys.variables.join(',');
+  const payload = {
+    ids, 
+    variable, 
+    datasetId: 'combined_e784_bee5_492e',
+    start: '2000-07-01T12:00:00Z',
+    end
+  };
+  return Promise.all(getMultiBuoyGeoJsonData(payload))
+    .then(
+      (response) => {
+        const data = response.map((datum) => {
+          console.log(datum);
+          return datum.data?.features.map((feature) => {
+            return feature.properties;
+          });
+        });
+
+        const rollupObject = {};
+        buoys.variables.forEach(v => {
+          rollupObject[v] = op.valid(v);
+        });
+
+        const processed = data.map((d) => {
+          let dt = aq.from(d)
+            .derive({
+              dt_ym: (d) =>
+                op.datetime(op.year(d.time), op.month(d.time)),
+              station_id: (d) => d.station_name
```

Comment:
might be more efficient to do the `station_name` => `station_id` rename with a `select`